javaslang/src/main/java/javaslang/Value.java

```diff
@@ -546,11 +546,10 @@ default CharSeq toCharSeq() {
                     componentType == long.class ? Long.class :
                     componentType == short.class ? Short.class :
                     componentType == void.class ? Void.class : null;
-            return toJavaArray((Class<T>) boxedType);
-        } else {
-            final java.util.List<T> list = toJavaList();
-            return list.toArray((T[]) java.lang.reflect.Array.newInstance(componentType, list.size()));
+            componentType = (Class<T>) boxedType;
         }
+        final java.util.List<T> list = toJavaList();
+        return list.toArray((T[]) java.lang.reflect.Array.newInstance(componentType, list.size()));
```

Comment:
now that we have `ArrayType`, we can create this without reflection (making it `GWT` compatible also).
Would be neet if you could JMH both, as it may also be faster than reflection :)